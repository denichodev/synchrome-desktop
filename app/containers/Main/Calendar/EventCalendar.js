// @flow

import React, { Component } from 'react';

import BigCalendar from 'react-big-calendar';
import moment from 'moment';

import events from './eventPlaceholder';

class EventCalendar extends Component {
  props: {
    height: number,
    width: number | '100%'
  }

  render() {
    BigCalendar.momentLocalizer(moment);
    const allViews = Object.keys(BigCalendar.views).map(k => BigCalendar.views[k])

    return (
      <div style={{
        height: this.props.height,
        width: this.props.width
      }}
      >
        <BigCalendar
          events={events}
          views={allViews}
          defaultDate={new Date(2015, 3, 1)}
        />
      </div>
    );
  }
}

export default EventCalendar;
