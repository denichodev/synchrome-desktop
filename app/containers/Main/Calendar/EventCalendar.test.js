import React from 'react';
import EventCalendar from './Calendar';

function setup() {
  const props = {};

  const enzymeWrapper = shallow(<EventCalendar {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('EventCalendar', () => {
  it('should match snapshot', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper).toMatchSnapshot();
  });
});
