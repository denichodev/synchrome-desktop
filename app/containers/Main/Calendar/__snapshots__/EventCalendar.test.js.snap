// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCalendar should match snapshot 1`] = `
<div
  className="animated fadein"
>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <div
        className="card card-inverse card-primary"
      >
        <div
          className="card-header"
        >
          Calendar
        </div>
        <div
          className="card-block"
        >
          <EventCalendar />
        </div>
      </div>
    </div>
  </div>
</div>
`;
